plugins/inputs/tomcat/tomcat.go

```diff
@@ -0,0 +1,176 @@
+package tomcat
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type TomcatStatus struct {
+	TomcatJvm        TomcatJvm         `xml:"jvm"`
+	TomcatConnectors []TomcatConnector `xml:"connector"`
+}
+
+type TomcatJvm struct {
+	JvmMemory      JvmMemoryStat       `xml:"memory"`
+	JvmMemoryPools []JvmMemoryPoolStat `xml:"memorypool"`
+}
+
+type JvmMemoryStat struct {
+	Free  int64 `xml:"free,attr"`
+	Total int64 `xml:"total,attr"`
+	Max   int64 `xml:"max,attr"`
+}
+
+type JvmMemoryPoolStat struct {
+	Name           string `xml:"name,attr"`
+	Type           string `xml:"type,attr"`
+	UsageInit      int64  `xml:"usageInit,attr"`
+	UsageCommitted int64  `xml:"usageCommitted,attr"`
+	UsageMax       int64  `xml:"usageMax,attr"`
+	UsageUsed      int64  `xml:"usageUsed,attr"`
+}
+
+type TomcatConnector struct {
+	Name        string      `xml:"name,attr"`
+	ThreadInfo  ThreadInfo  `xml:"threadInfo"`
+	RequestInfo RequestInfo `xml:"requestInfo"`
+}
+
+type ThreadInfo struct {
+	MaxThreads         int64 `xml:"maxThreads,attr"`
+	CurrentThreadCount int64 `xml:"currentThreadCount,attr"`
+	CurrentThreadsBusy int64 `xml:"currentThreadsBusy,attr"`
+}
+type RequestInfo struct {
+	MaxTime        int   `xml:"maxTime,attr"`
+	ProcessingTime int   `xml:"processingTime,attr"`
+	RequestCount   int   `xml:"requestCount,attr"`
+	ErrorCount     int   `xml:"errorCount,attr"`
+	BytesReceived  int64 `xml:"bytesReceived,attr"`
+	BytesSent      int64 `xml:"bytesSent,attr"`
+}
+
+type Tomcat struct {
+	URL      string
+	Username string
+	Password string
+}
+
+var sampleconfig = `
+  ## A Tomcat status URI to gather stats.
+  ## Default is "http://127.0.0.1:8080/manager/status/all?XML=true".
+  url = http://127.0.0.1:8080/manager/status/all?XML=true
+	## Cedentials for status URI.
+	## Default is tomcat/s3cret.
+	username = "tomcat"
+	password = "s3cret"
```

Comment:
Fix up this indention because it is used directly in the sample config output.  You can test with either of these commands:
```
telegraf --usage tomcat
telegraf config
```